src/amplitude-client.js

```diff
@@ -1633,6 +1688,23 @@ AmplitudeClient.prototype.sendEvents = function sendEvents() {
     checksum: md5(Constants.API_VERSION + this.options.apiKey + events + uploadTime),
   };
 
+  if (!navigator.sendBeacon) {
+    this.options.transport = Constants.TRANSPORT_HTTP;
+  }
+
+  if (this.options.transport === Constants.TRANSPORT_BEACON) {
```

Comment:
might be too much, but would be good to replace a lot of these with a small descriptive helper function like `doesTransportSupportSaveEvents`

Reply:
I tried adding helper functions, but since there are only two transports now, they didn't add much descriptivity compared to checking the value itself. If there end up being more than two, then definitely there should be helper functions to categorize them like this